source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -89,10 +116,20 @@ void DnsResolverImpl::initializeChannel(ares_options* options, int optmask) {
     static_cast<DnsResolverImpl*>(arg)->onAresSocketStateChange(fd, read, write);
   };
   options->sock_state_cb_data = this;
-  ares_init_options(&channel_, options, optmask | ARES_OPT_SOCK_STATE_CB);
+  optmask |= ARES_OPT_SOCK_STATE_CB;
+
+  ENVOY_LOG_MISC(error, "resolvers {}, use dns fallback {}", resolvers_vector_.size(),
```

Comment:
nit I think this should be warn, same below. Also, perhaps this should be deleted in place of the log in that if block below?

Reply:
Sorry, this log line and the one below are actually just debug statements I was working with. They appeared back after a bad merge. Cleaning.